src/components/switchTheme/useSwitchTheme.tsx

```diff
@@ -0,0 +1,56 @@
+import { createContext, ReactNode, useCallback, useContext, useEffect, useState } from 'react';
+
+interface ThemeContextType {
+  theme: 'light' | 'dark';
+  toggleTheme: () => void;
+}
+
+interface IProps {
+  children: ReactNode;
+}
+
+export const ThemeContext = createContext<ThemeContextType | null>(null);
+
+const ThemeContextProvider = ({ children }: IProps) => {
+  const getInitialTheme = () => {
+    const savedTheme = localStorage.getItem('theme');
+    if (savedTheme) {
+      return savedTheme as 'light' | 'dark';
+    }
+
+    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+    return prefersDark ? 'dark' : 'light';
+  };
+
+  const [theme, setTheme] = useState<'light' | 'dark'>(getInitialTheme);
+
+  const toggleTheme = useCallback(() => {
+    setTheme((prev) => {
+      const newTheme = prev === 'light' ? 'dark' : 'light';
+      document.documentElement.setAttribute('data-theme', newTheme);
+      localStorage.setItem('theme', newTheme);
+      return newTheme;
+    });
+  }, []);
+
+  useEffect(() => {
+    document.documentElement.setAttribute('data-theme', theme);
+  }, [theme]);
+
+  const value = {
+    theme,
+    toggleTheme,
+  };
+
+  return <ThemeContext.Provider value={value}>{children}</ThemeContext.Provider>;
+};
+
+export default ThemeContextProvider;
+
+export const useTheme = () => {
+  const context = useContext(ThemeContext);
+  if (context === null) {
+    throw new Error('Context must be used within a context provider');
+  }
+  return context;
+};
```

Comment:
why one of the exports is default?

Reply:
I changed but for example App component has also export default

Reply:
I'm just trying to understand whether you understand what you wrote. Seems like not really. Please, prepare answer to all this questions before our next call.